pypeit/core/flexure.py

```diff
@@ -732,22 +731,29 @@ def sky_em_residuals(wave:np.ndarray, flux:np.ndarray,
     diff_err  = []
     los = []
     los_err= []
+
+    good_ivar = ivar > 0
+
     # Loop on known sky lines
     for line in sky_waves: 
         wline = [line-noff,line+noff] 
-        mw    = (wave > wline[0]) & (wave < wline[1])
+        mw    = (wave > wline[0]) & (wave < wline[1]) & good_ivar
         
         # Reuire minimum number
         if np.sum(mw) <= nfit_min:
             continue
 
         p=[0,0,0,0]
         # Guess
-        p0 = list(fitting.guess_gauss(wave[mw],flux[mw]))
+        p0 = list(fitting.guess_gauss(wave[mw], flux[mw]))
         # Fit
-        p, pcov = fitting.fit_gauss(wave[mw],flux[mw], 
-                                w_out = 1./np.sqrt(ivar[mw]), 
-                                guesses=p0, nparam=4)
+        try:
```

Comment:
Please add a failure specific except statement here, not a general except, which is not very good coding practice. It looks like you know what failure to expect here, namely the fit crashes out.

Reply:
Done.